src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -878,22 +881,24 @@ void readAnalyze(TrackPointer track, double sampleRate, int timingOffset, bool i
                 case rekordbox_anlz_t::CUE_LIST_TYPE_MEMORY_CUES: {
                     switch ((*cueEntry)->type()) {
                     case rekordbox_anlz_t::CUE_ENTRY_TYPE_MEMORY_CUE: {
-                        memory_cue_t memoryCue;
-                        memoryCue.position = position;
-                        memoryCue.color = mixxx::RgbColor::nullopt();
-                        memoryCues << memoryCue;
+                        memory_cue_loop_t memoryCue;
+                        memoryCue.startPosition = position;
+                        memoryCue.endPosition = -1;
+                        memoryCue.colorID = -1;
+                        memoryCuesLoops << memoryCue;
                     } break;
                     case rekordbox_anlz_t::CUE_ENTRY_TYPE_LOOP: {
-                        // As Mixxx can only have 1 saved loop, use the first occurrence of a memory loop relative to the start of the track
-                        if (position < cueLoopStartPosition) {
-                            cueLoopStartPosition = position;
-                            int endTime = static_cast<int>((*cueEntry)->loop_time()) - timingOffset;
-                            // Ensure no offset times are less than 1
-                            if (endTime < 1) {
-                                endTime = 1;
-                            }
-                            cueLoopEndPosition = samples * static_cast<double>(endTime);
+                        int endTime = static_cast<int>((*cueEntry)->loop_time()) - timingOffset;
+                        // Ensure no offset times are less than 1
+                        if (endTime < 1) {
+                            endTime = 1;
                         }
+
+                        memory_cue_loop_t loop;
+                        loop.startPosition = position;
+                        loop.endPosition = samples * static_cast<double>(endTime);
+                        loop.colorID = -1;
```

Comment:
Rekordbox has no color for loops? If so, please document that with a comment here.

Reply:
It does, this section of code is for legacy cues which lack color information. Can add a comment reflecting this.